main.py

```diff
@@ -246,16 +248,34 @@ def search_exercises():
     )
 
 
-@main.route("/search_workouts")
+@main.route("/search_workouts", methods=("GET", "POST"))
 @login_required
 def search_workouts():
     """Search workouts page."""
     db_utils = SqliteUtilites(DATABASE_NAME)
-    workouts = db_utils.execute(
-        "SELECT a.name, a.description, a.id, a.created, b.name AS author FROM"
-        " workouts a JOIN users b ON a.user_id = b.id",
-        fetch_all=True,
-    )
+    
+    if request.method == "POST":
+        workout_name = sanitize_input(request.form.get("workout_name"))
+        if not workout_name:
+            workouts = db_utils.execute(
+                "SELECT a.name, a.description, a.id, a.created, b.name AS author FROM"
+                " workouts a JOIN users b ON a.user_id = b.id",
+                fetch_all=True,
+            )
+        else:
+            workouts = db_utils.execute(
+                "SELECT a.name, a.description, a.id, a.created, b.name AS author FROM"
+                " workouts a JOIN users b ON a.user_id = b.id"
+                f" WHERE a.name = '{workout_name}'",
```

Comment:
This is a very useful feature. 

One suggestion just to improve the utility is that we could like for similar workout names instead of literal names.
i.e.
Instead of  "WHERE a.name = '{workout_name}'", something like "WHERE a.name LIKE '%{workout_name}%'"